### PR TITLE
refactor: Table render logic opt

### PR DESCRIPTION
**Pull request by zombieJ**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [x] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 💡 Background and solution

Since `column.render` return `props` still legacy support in v5, legacy check have to keep here. But we use `rc-component/context` to enhance render times check. Follow is compare 500 items with check state update (control 2x+ & uncontrol 4x+ render perf enhance):

#### Control Before

截屏2023-01-06 16 36 31

#### Control After

截屏2023-01-06 16 32 29

#### Uncontrol Before

截屏2023-01-06 17 03 35

#### Uncontrol After

截屏2023-01-06 16 56 32

#### What Changed

* Move file `Table.tsx` to `InternalTable.tsx`
* Create `Table.tsx` record parent render times and pass prop to RcTable.
* Create `RcTable.tsx` for `genTable` function to handle `Table.tsx` parent render logic check.

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |    Optimize Table logic to reduce `shouldCellUpdate` unnecessary rendering times(Enhance 100% render perf on control and 400% perf on un-control)       |
| 🇨🇳 Chinese |      优化 Table 逻辑，减少 `shouldCellUpdate` 不必要的渲染次数（提升受控下 100% 性能，非受控 400% 性能）。     |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

**Review by heiyu4585 — APPROVED** [review]

File: components/table/InternalTable.tsx [hunk 1]
```diff
@@ -0,0 +1,553 @@
...
+  };
+
+  // ========================== Expandable ==========================
+
+  // Pass origin render status into `rc-table`, this can be removed when refactor with `rc-table`
+  (mergedExpandable as any).__PARENT_RENDER_ICON__ = mergedExpandable.expandIcon;
+
+  // Customize expandable icon
+  mergedExpandable.expandIcon =
+    mergedExpandable.expandIcon || expandIcon || renderExpandIcon(tableLocale!);
```
Comment (kiner-tang) [inline components/table/InternalTable.tsx:435]: 这里是不是用`??`会好点，如果用户明确传了`null`进来呢？

Comment (zombieJ) [reply components/table/InternalTable.tsx:435]: 这部分代码没有重写，只是把 Table 挪到 InternalTable 文件里。我更新一下主要修改部分到 description 里。稍等。

Comment (zombieJ) [reply components/table/InternalTable.tsx:435]: Update in `What Changed`

**Comment by afc163** [comment]
Changelog 里可以给个性能提升了多少的效果描述，以及对比 codesandbox

**Review by li-jia-nan — APPROVED** [review]

**Comment by zombieJ** [comment]
> Changelog 里可以给个性能提升了多少的效果描述，以及对比 codesandbox

Changelog Updated。Codesandbox 得发个版本搞之，另外 500 条数据几百 ms 的差异体感上不会特别明显，放上千条性能消耗又会转移到重绘上也感受不到。

**Comment by afc163** [comment]
> 提升受控下 100% 性能，非受控 400% 性能

这是使用了 shouldUpdate 的场景才会提升？还似乎默认情况下也会提升？

提升的是首次 render 渲染性能还是交互后 rerender 的性能？

**Comment by zombieJ** [comment]
> 这是使用了 shouldUpdate 的场景才会提升？还似乎默认情况下也会提升？

默认有闭包在，Cell 总是要全部渲染的。 ref #40004

**Comment by afc163** [comment]
> 默认有闭包在，Cell 总是要全部渲染的。 ref https://github.com/ant-design/ant-design/pull/40004

好像没回答问题。

**Comment by afc163** [comment]
codesandbox 对比可以给一个了

**Pull request merged**